### PR TITLE
Change schema to use name and models to use id

### PR DESCRIPTION
**Pull request by EspenK**
Change some stuff that was done in 22374a50943d16e710eba49359915523258fe54f.

### CODE REVIEW

**Review by pckv — APPROVED** [review]
Looks good. The database should use the ID as reference, even if the API schema identifies by name for User and Resub. Conversion can be done in the CRUD functions by using the relationships in the model.

File: repost/models/comment.py [hunk 1]
```diff
@@ -12,11 +12,10 @@ class Comment(Base):
     created = Column(DateTime)
     edited = Column(DateTime, nullable=True)
 
-    author_name = Column(String, ForeignKey('users.id'))
-    parent_resub_name = Column(String, ForeignKey('resubs.name'))
-    parent_post_name = Column(String, ForeignKey('posts.title'))
-    parent_comment_id = Column(Integer, ForeignKey('comment.id'), nullable=True)
-    votes = Column(Integer)
+    author_id = Column(String, ForeignKey('users.id'))
```
Comment (jonsondrem) [inline repost/models/comment.py]: Change from String to Integer

File: repost/models/comment.py [hunk 2]
```diff
@@ -12,11 +12,10 @@ class Comment(Base):
...
+    parent_resub_id = Column(String, ForeignKey('resubs.id'))
```
Comment (jonsondrem) [inline repost/models/comment.py]: Change from String to Integer

File: repost/models/comment.py [hunk 3]
```diff
@@ -12,11 +12,10 @@ class Comment(Base):
...
+    parent_post_id = Column(String, ForeignKey('posts.id'))
```
Comment (jonsondrem) [inline repost/models/comment.py]: Change from String to Integer

File: repost/models/post.py [hunk 4]
```diff
@@ -13,10 +13,9 @@ class Post(Base):
     content = Column(String, nullable=True)
     created = Column(DateTime)
     edited = Column(DateTime, nullable=True)
-    votes = Column(Integer)
 
-    author_name = Column(String, ForeignKey('users.username'))
-    parent_resub_name = Column(String, ForeignKey('resubs.name'))
+    author_id = Column(String, ForeignKey('users.id'))
```
Comment (jonsondrem) [inline repost/models/post.py]: Change from String to Integer

File: repost/models/post.py [hunk 5]
```diff
@@ -13,10 +13,9 @@ class Post(Base):
...
+    parent_resub_id = Column(String, ForeignKey('resubs.id'))
```
Comment (jonsondrem) [inline repost/models/post.py]: Change from String to Integer

**Review by jonsondrem — CHANGES_REQUESTED** [review]
Some strings need to be changed to Integer if they are to be changed from name to id

**Comment by pckv** [comment]
> Some strings need to be changed to Integer if they are to be changed from name to id

Nice catch! Fixed with 2811cd9d7cd869f78b5cb4c6afd5bf37c3094b2e

**Review by jonsondrem — APPROVED** [review]
:+1:

**Pull request merged**